src/Npgsql.SourceGenerators/GeneratedDatabaseInfo.snbtxt

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Npgsql.Internal;
+using Npgsql.PostgresTypes;
+using Npgsql.Util;
+
+#nullable enable
+
+namespace Npgsql;
+
+partial class GeneratedDatabaseInfo
+{
+    static readonly Lazy<PostgresType[]> Types = new(() =>
```

Comment:
What's the reason for this needing to be lazy?

Reply:
That's a leftover from the time I wasn't sure about our instantiation behavior.
I'll remove it.

Reply:
Unless I'm mistaken, you should be able to have a simple `static readonly PostgresType[] Types` in the non-generated GeneratedDatabaseInfo.cs, and source-generate a static constructor with the same code you have now (but directly initializing Types at the end - no need for Lazy<>).

Reply:
Yeah a static constructor would work too but not a simple array initializer.
I'll move it to as static constructor.